protocol_driver_mercury.h

```diff
@@ -78,6 +78,8 @@ class ProtocolDriverMercury : public ProtocolDriver {
 
   hg_context_t* hg_context_ = nullptr;
   hg_class_t* hg_class_ = nullptr;
+  static constexpr bool mercury_ipv4_only_ =
+      (HG_VERSION_MAJOR < 2) || (HG_VERSION_MAJOR == 2 && HG_VERSION_MINOR < 2);
```

Comment:
This makes sense now. HG_VERSION_MAJOR /MINOR determines this bool which determines the netdev_name above. Is that correct?

Reply:
It determines if the protocol driver asks for IP version 4, or allows for any IP version to be used.